libautoscoper/src/Trial.cpp

```diff
@@ -195,47 +338,54 @@ namespace xromm
         cerr << e.what() << endl;
       }
     }
+    num_frames = maxVideoFrames;
+  }
 
+  void Trial::loadOffsets(std::vector<std::string>* optimizationOffsets) {
     // Read in the offsets, otherwise default to 0.1
     offsets[0] = 0.1; offsets[1] = 0.1; offsets[2] = 0.1;
     offsets[3] = 0.1; offsets[4] = 0.1; offsets[5] = 0.1;
-    if (!optimizationOffsets.empty()) {
-      stringstream offset_stream(optimizationOffsets.back());
+    if (!optimizationOffsets->empty()) {
+      stringstream offset_stream(optimizationOffsets->back());
       offset_stream >> offsets[0] >> offsets[1] >> offsets[2] >>
         offsets[3] >> offsets[4] >> offsets[5];
     }
+	}
 
+  void Trial::loadRenderResolution(std::vector<std::string>* renderResolution) {
     // Read in the rendering dimensions, default to 512
     render_width = 512;
     render_height = 512;
-    if (!renderResolution.empty()) {
-      stringstream resolution_stream(renderResolution.back());
+    if (!renderResolution->empty()) {
+      stringstream resolution_stream(renderResolution->back());
       resolution_stream >> render_width >> render_height;
     }
-
-    // Initialize the coordinate frames
-    num_frames = maxVideoFrames;
-  }
+	}
 
   void Trial::save(const std::string& filename)
   {
+
+    fs::path configDir = fs::path(filename).parent_path();
+
     ofstream file(filename.c_str());
     if (!file) {
       throw runtime_error("Failed to save to file: " + filename);
     }
 
     file.precision(12);
 
+    file << "Version 1.1" << endl;
+
     for (unsigned i = 0; i < cameras.size(); ++i) {
-      file << "mayaCam_csv " << cameras.at(i).mayacam() << endl;
+      file << "mayaCam_csv " << fs::relative(cameras.at(i).mayacam(), configDir).string() << endl;
     }
 
     for (unsigned i = 0; i < videos.size(); ++i) {
-      file << "CameraRootDir " << videos.at(i).dirname() << endl;
+      file << "CameraRootDir " << fs::relative(videos.at(i).dirname(),configDir).string() << endl;
     }
 
     for (unsigned i = 0; i < volumes.size(); ++i) {
-      file << "VolumeFile " << volumes.at(i).name() << endl;
+      file << "VolumeFile " << fs::relative(volumes.at(i).name(),configDir).string() << endl;
```

Comment:
```suggestion
      file << "VolumeFile " << fs::relative(volumes.at(i).name(), configDir).string() << endl;
```

Adding a space between parameters improve readability